### PR TITLE
Data dictionaries functions

### PR DESCRIPTION
**Pull request by aspina7**
This is a first pass at working with the msf data dictionaries. 
TLDR: WIP :) 

Contains two functions: one for pulling dictionary, second for generating fake data. 

I am sure that my use of apply and sample can be greatly improved - but hey it works... 

We are waiting to receive newer data dictionaries from HQ (depending on their work load begin of week), which will hopefully include: 
1) Front end - what data entry mask looks like
2) Back end - corresponding to data entry but what shows what excel export looks like 
3) Rules - restrictions and variable dependencies 

From those, would need to change the data dictionary function to return a list (i guess?) with the three data sets mentioned above in. 

Theres also a lot that can be improved in the fake datasets, to make them more realistic - like making date of exit later than date of entry for example. But hopefully that will come with the rules dataset above. (Also hopefully some of the data manipulation can be removed once the data dictionaries are more structured). 

Hopefully this isnt too shamefully coded :)

### CODE REVIEW

**Comment by aspina7** [comment]
also @zkamvar the huxtable table in the outbreak sitrep comes out like this and i cant seem to fix it

image

**Comment by zkamvar** [comment]
> also @zkamvar the huxtable table in the outbreak sitrep comes out like this and i cant seem to fix it
> 
> image

I believe this is fixed by setting the `results = 'asis'` in the chunk header.

File: R/gen_data.R [hunk 1]
```diff
@@ -1,53 +1,159 @@
...
+  # clean col names
+  colnames(dat_dict) <- epitrix::clean_labels(colnames(dat_dict))
+
+  # clean future var names (shortname_export)
+  dat_dict$shortname_export <- epitrix::clean_labels(dat_dict$shortname_export)
+
+  # DATA CLEANING
+  # (CAN EVENTUALLY BE DELETED WHEN DICTIONARIES STABILISE)
+  # 3 types added to - BOOLEAN, TRUE_ONLY and ORGANISATION-UNIT
+  dat_dict$options[which(is.na(dat_dict$options) &
```
Comment (zkamvar) [inline R/gen_data.R]: You don't actually need the "which" function here; the logical subsetting will take care of things

File: R/gen_data.R [hunk 2]
```diff
@@ -1,53 +1,159 @@
...
+  dat_dict$x_1[which(is.na(dat_dict$x_1) &
+                       dat_dict$value_type == "TRUE_ONLY")] <- "NA"
+
+
+
+  dat_dict$options[which(is.na(dat_dict$options) &
+                           dat_dict$value_type == "ORGANISATION_UNIT")] <- "Hospital"
+  dat_dict$x_1[which(is.na(dat_dict$x_1) &
+                       dat_dict$value_type == "ORGANISATION_UNIT")] <- "Clinic"
+  dat_dict$x_1[which(is.na(dat_dict$x_1) &
```
[the remark this answers is not in the document]
Comment (zkamvar) [reply R/gen_data.R]: Additionally, this particular line overwrites the previous line

**Comment by aspina7** [comment]
Merging for now - we are awaiting more structured linelists. 
We will work on it from here and then new PR once updates come in
